api/policies/sessionAuth.js

```diff
@@ -1,8 +1,40 @@
-module.exports = function(req, res, next) {
-  req.session.authRedirectPath = undefined
+const moment = require('moment');
+const GitHub = require('../services/GitHub');
+
+// TODO: Move this to another file and export it for easier testing
+function isAuthDatePastThreshold(authDate, options = {}) {
+  const threshAmount = options.threshAmount || 30;
+  const threshUnits = options.threshUnits || 'minutes';
+  return moment().isAfter(moment(authDate).add(threshAmount, threshUnits));
+}
+
+// TODO: new tests
+module.exports = function sessionAuth(req, res, next) {
+  req.session.authRedirectPath = undefined;
   if (req.session.authenticated) {
+    const lastAuthenticatedAt = req.session.authenticatedAt;
+    if (!lastAuthenticatedAt || isAuthDatePastThreshold(lastAuthenticatedAt)) {
```

Comment:
the `!lastAuthenticatedAt` check is so this doesn't bomb out for existing active user sessions. rather, if missing it will do the `validateUser` check and add the field if that check passes.